src/components/OSCALLoader.js

```diff
@@ -140,7 +140,9 @@ export default function OSCALLoader(props) {
   };
 
   const handleUrlChange = (value) => {
-    setOscalUrl(value);
+    let tempValue;
+    if (value.contains("#")) tempValue = value.split("#").pop();
+    setOscalUrl(tempValue);
```

Comment:
```suggestion
    if (value.contains("#")) {
      setOscalUrl(value.split("#").pop());
    }
    setOscalUrl(undefined);
```

Is it actually intentional that this can be set to `undefined` in so may situations? Also should we _really_ be pulling this by splitting on `"#"`? Or should we use the `URL()` constructor and [`hash`](https://developer.mozilla.org/en-US/docs/Web/API/URL/hash)

Reply:
I'll admit this part is a bit rough and unfinished. I was investigating why there was a 404 error in the console when refreshing the page with a provided hash.

The idea here is that if the URL contains a hash it is removed from the URL, otherwise the URL would be handled normally. In the case of a URL containing a hash, we still want to keep track of the hash to navigate to after the page loads, so it is stored in `tempValue`. I was planning to migrate some of the code written in the `useEffect()` below into it's own constant, to be called at the end of this function with `tempValue` as it's parameter.

As you mentioned, I think `hash` would make more sense to handle and exclude here.